### ISSUE TITLE
Foreach: UnusedLocalVariable $value (since #48) is closed

### ISSUE THREAD

**Issue by marcelloh** [phpmd/phpmd#424]
Hi,

I'm getting error (Avoid unused local variables such as '$value'.) on this kind of code.
```
foreach($array as $key => $value)
{
   doSomething($key);
}
```
I think this is still a bug

There is change:
Allow unused foreach variables #198
but it is either not in the l;atest lease, or it is not fixing what it should.

**Comment by MacSim75** [comment]
Anything new here or should we @SuppressWarnings(PHPMD.UnusedLocalVariable) which is pretty ugly btw

**Comment by ravage84** [comment]
@marcelloh  & @MacSim75 have you both set the ``allow-unused-foreach-variables`` property to true for the UnusedLocalVariable rule? If so, how did you set it?

**Comment by marcelloh** [comment]
```
<rule ref="rulesets/unusedcode.xml/UnusedLocalVariable">
    <properties>
        <property name="allow-unused-foreach-variables" value="true"/>
    </properties>
	<priority>3</priority>
</rule>
```

**Comment by tanael** [comment]
I've had the same problem (version 2.4.3-1 on Debian Stretch).

I figured out that trying to overload the `allow-unused-foreach-variables` property value to `true` through a rule reference (as described in documentation https://phpmd.org/documentation/creating-a-ruleset.html) does not work. You have to either set it to `true` inside `unusedcode.xml` directly or copy the entire rule into your custom ruleset and set it there.

**Comment by DanielSchwiperich** [comment]
It works fine when doing it this way:

phpmd-ruleset.xml
```xml
 <rule ref="rulesets/unusedcode.xml">
    <exclude name="UnusedLocalVariable"/>
</rule>
<rule ref="rulesets/unusedcode.xml/UnusedLocalVariable">
    <properties>
        <property name="allow-unused-foreach-variables" value="true" />
    </properties>
</rule>
```

I think this issue can be closed

**Comment by ravage84** [comment]
Thank you @danielschwiperich
